grails-app/controllers/org/pih/warehouse/order/PurchaseOrderWorkflowController.groovy

```diff
@@ -54,13 +54,18 @@ class PurchaseOrderWorkflowController {
         enterOrderDetails {
             on("next") {
                 log.info "Save order details " + params
-                flow.order.properties = params
-                try {
-                    if (!orderService.saveOrder(flow.order)) {
+                if (flow.order.orderItems && flow.order.origin.id != params.origin.id) {
```

Comment:
The one issue I have with this is that it's not going to be applied whenever we save an order. It would only be applied when we run through this exact code. So I'm going to work on adding this as a constraint to the Order class.